types/react-datepicker/index.d.ts

```diff
@@ -72,14 +74,18 @@ export interface ReactDatePickerProps {
     nextMonthButtonLabel?: string;
     nextYearButtonLabel?: string;
     onBlur?(event: React.FocusEvent<HTMLInputElement>): void;
+    onCalendarClose?(...args: any): void;
+    onCalendarOpen?(...args: any): void;
```

Comment:
What are these `args` used for?

Reply:
Those functions are called [here](https://github.com/Hacker0x01/react-datepicker/blob/master/src/index.jsx#L259). The [provided `PropType` indicates those two](](https://github.com/Hacker0x01/react-datepicker/blob/master/src/index.jsx#L162)) are of type `PropTypes.func`, so I went to the type definitions for `prop-types` to get that [definition](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/prop-types/index.d.ts#L61).

I imagine these lines could be changed to:
```
onCalendarClose?(): void;
onCalendarOpen?(): void;
```
... as `react-datepicker` itself doesn't provide a callback signature, but I left it as `...args` in case the function assigned to either accepted arguments.

Reply:
I'd change them to what you suggested. `react-datepicker` doesn't provide any arguments for the callbacks and the types should reflect that.

Reply:
I'll get that updated this afternoon and pushed over.

Reply:
I've updated the definitions to reflect suggested changes with commit 7b16773b145f8fd9c2fa58b5932178bd951c86c3 below. @tu4mo

Reply:
LGTM 👍